tasks/label_analysis.py

```diff
@@ -101,6 +106,10 @@ async def run_async(self, db_session, request_id, *args, **kwargs):
                 commit=label_analysis_request.head_commit.commitid,
             ),
         )
+        self.metrics_context = MetricContext(
```

Comment:
i don't think celery wants us to store per-request state on the task like this https://docs.celeryq.dev/en/latest/userguide/tasks.html#instantiation. we probably should either recreate the context or pass it around as a function argument, though i don't really like that :P

Reply:
Oh that is a very good observation.

What do you mean by "recreate the context"? We are doing that, no? Since the entrypoint for the task is `run_async` and we create a new context every time.

Reply:
recreate in each function where we need it, i mean